vm/vm.go

```diff
@@ -188,37 +196,29 @@ func (v *vm) Execute(txns []core.Transaction, declaredClasses []core.Class, bloc
 	paidFeesOnL1CStr := cstring(paidFeesOnL1Bytes)
 	txnsJSONCstr := cstring(txnsJSON)
 	classesJSONCStr := cstring(classesJSON)
+	contractInfoJSONCStr := cstring(contractInfoJSON)
 
 	sequencerAddressBytes := sequencerAddress.Bytes()
-	gasPriceWEIBytes := gasPriceWEI.Bytes()
 
-	if gasPriceSTRK == nil {
-		gasPriceSTRK = &felt.Zero
-	}
-	gasPriceSTRKBytes := gasPriceSTRK.Bytes()
+	gasPriceWEIBytes := getBytesOrSetDefault(gasPriceWEI)
+	gasPriceSTRKBytes := getBytesOrSetDefault(gasPriceSTRK)
+	daGasPriceWEIBytes := getBytesOrSetDefault(daGasPriceWEI)
+	daGasPriceFRIBytes := getBytesOrSetDefault(daGasPriceFRI)
 
-	var skipChargeFeeByte byte
-	if skipChargeFee {
-		skipChargeFeeByte = 1
+	if gasPriceWEI.IsZero() || gasPriceSTRK.IsZero() || daGasPriceWEI.IsZero() || daGasPriceFRI.IsZero() {
+		return nil, nil, errors.New("gas prices must be non-zero")
```

Comment:
Since these are not user provided, it is fine to panic and crash in Rust if these are zero.